### DISCUSSION TITLE
How to make robot dock/undock in Python

### DISCUSSION THREAD

**Discussion by poojaravin** [iRobotEducation/create3_docs#418] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

HP Elitebook, Raspberry Pi/Ubuntu 20.04

### Which version of ROS 2 is installed on your computer?

Galactic

### Which firmware version is installed on your robot?

G.4.5

### Which RMW is your robot running?

CycloneDDS

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

I have the ROS2 with my raspberry pi, I would want to code using python for my project. 
Can somebody help me on where i can start and some basic examples like docking and undocking in python?

**Comment by kwujciak** [comment, marked as answer]
[Here](https://github.com/tuftsceeo/Tufts_Create3_Examples/blob/main/Code/Package/example_package/action_dock.py) is a docking example I've played around with, and [this](https://github.com/tuftsceeo/Tufts_Create3_Examples/blob/main/Code/Package/example_package/action_undock.py) is an undocking example I've used. Both use rclpy.
